src/inputs/TagsEditor/TagsEditor.js

```diff
@@ -0,0 +1,191 @@
+// Copyright (c) Cosmo Tech.
+// Licensed under the MIT license.
+
+import React, { useCallback, useMemo, useState } from 'react';
+import PropTypes from 'prop-types';
+import { Chip, Grid, TextField, Typography } from '@mui/material';
+import makeStyles from '@mui/styles/makeStyles';
+import EditIcon from '@mui/icons-material/Edit';
+
+const DEFAULT_LABELS = {
+  header: 'Tags',
+  placeholder: 'Enter a new tag',
+};
+
+const useStyles = makeStyles((theme) => ({
+  label: {
+    color: theme.palette.text.secondary,
+  },
+  addTagIcon: {
+    width: '16px',
+    height: '16px',
+  },
+}));
+
+export const TagsEditor = (props) => {
+  const classes = useStyles();
+  const { id, values, readOnly, onChange, labels: tmpLabels, chipProps, textFieldProps, editIconProps } = props;
+  const labels = { ...DEFAULT_LABELS, ...tmpLabels };
+
+  const [isEditing, setIsEditing] = useState(false);
+  const [isHovered, setIsHovered] = useState(false);
+
+  const startEdition = useCallback((event) => {
+    event.stopPropagation();
+    event.preventDefault();
+    setIsEditing(true);
+  }, []);
+
+  const stopEdition = useCallback((event) => {
+    event.stopPropagation();
+    event.preventDefault();
+    setIsEditing(false);
+  }, []);
+
+  const addNewTag = useCallback(
+    (newTag) => {
+      setIsEditing(false);
+      if (newTag.trim().length === 0) return;
+      onChange([...values, newTag]);
+    },
+    [onChange, values]
+  );
+
+  const deleteTag = useCallback(
+    (removedTagIndex) => {
+      const newTags = [...values];
+      newTags.splice(removedTagIndex, 1);
+      onChange(newTags);
+    },
+    [onChange, values]
+  );
+
+  const newTagInput = useMemo(() => {
+    if (readOnly || !isEditing) return null;
+    return (
+      <TextField
+        data-cy="new-tag-textfield"
+        sx={{ maxWidth: '150px' }}
```

Comment:
Maybe, we can leave the width not defined, so the component will fit it's parent's width by default and a specific width value could be passed via `textFieldProps` ? :slightly_smiling_face:

Reply:
@esasova  I don't know, usually, we set a maximum width to all `TextFields` to prevent them from taking too much space horizontally, even if the parent has free space. The default value of `150px` seemed appropriate to fit the average size of a dataset tag (so the width of the input field is consistent with the size of chips on the same line), and this default value can still be overwritten by `textFieldProps` if we want it to be `100%` in some other context for example.
What do you think ? :slightly_smiling_face: